### PR TITLE
[Feature] Support histogram statistics syntax and AST

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 35 / 47 (74.47%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AnalyzeHistogramDesc.java|5|10|50.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|13|18|72.22%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|11|13|84.62%|
|:large_blue_circle:|com/starrocks/sql/ast/AnalyzeBasicDesc.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AnalyzeStmt.java|3|3|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java [hunk 1]
```diff
@@ -138,5 +142,22 @@ public Void visitCreateAnalyzeJobStatement(CreateAnalyzeJobStmt statement, Conne
...
         }
+
+        void analyzeAnalyzeTypeDesc(ConnectContext session, AnalyzeStmt statement, AnalyzeTypeDesc analyzeTypeDesc) {
+            if (analyzeTypeDesc instanceof AnalyzeHistogramDesc) {
+                List<String> columns = statement.getColumnNames();
+                OlapTable analyzeTable = (OlapTable) MetaUtils.getStarRocksTable(session, statement.getTableName());
+
+                for (String columnName : columns) {
+                    Column column = analyzeTable.getColumn(columnName);
+                    if (column.getType().isComplexType()
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java:153]: How about move this if statement to `Type.java`, and naming it as `Type::supportHistogram`?

File: fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AnalyzeStmtTest.java [hunk 2]
```diff
@@ -55,4 +56,12 @@ public void testProperties() {
         Assert.assertEquals(1, analyzeStmt.getProperties().size());
         Assert.assertEquals("30", analyzeStmt.getProperties().getOrDefault("expire_sec", "2"));
     }
+
+    @Test
+    public void testHistogram() {
+        String sql = "analyze table t0 update histogram on v1,v2 with 256 buckets";
```
Comment (mofeiatwork) [inline fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AnalyzeStmtTest.java:62]: This statement is so complex, does our user need to use it manually?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java (hunk 1 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java:153]: I think these types of judgments are abstract enough, and they can be used for combination. It is not necessary to put all type combinations into Type.java

File: fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AnalyzeStmtTest.java (hunk 2 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AnalyzeStmtTest.java:62]: 1. Histogram calculation is a costly operation, and the general database is triggered manually. 
2. This syntax is compatible with mysql's histogram syntax

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 3]
```diff
@@ -1278,6 +1278,18 @@ public class Config extends ConfigBase {
     @ConfField(mutable = true)
     public static boolean enable_statistic_collect = true;
 
+    /**
+     * default bucket size of histogram statistics
+     */
+    @ConfField(mutable = true)
+    public static long histogram_buckets_size = 256;
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:1285]: why not use session Variable?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java [hunk 4]
```diff
@@ -138,5 +142,22 @@ public Void visitCreateAnalyzeJobStatement(CreateAnalyzeJobStmt statement, Conne
             }
             return null;
         }
+
+        void analyzeAnalyzeTypeDesc(ConnectContext session, AnalyzeStmt statement, AnalyzeTypeDesc analyzeTypeDesc) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java:146]: This method name is a little strange.

Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java:146]: give a suggested name？

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java (hunk 4 above)
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java:146]:
```suggestion
        private void analyzeAnalyzeTypeDesc(ConnectContext session, AnalyzeStmt statement, AnalyzeTypeDesc analyzeTypeDesc) {
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 35 / 47 (74.47%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AnalyzeHistogramDesc.java|5|10|50.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|13|18|72.22%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|11|13|84.62%|
|:large_blue_circle:|com/starrocks/sql/ast/AnalyzeBasicDesc.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AnalyzeStmt.java|3|3|100.00%|

**Review by Youngwb — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**